StoreMessageContentActivity/handler.ts

```diff
@@ -53,13 +67,107 @@ export type StoreMessageContentActivityResult = t.TypeOf<
   typeof StoreMessageContentActivityResult
 >;
 
+export const ServicePreferenceError = t.interface({
+  kind: t.keyof({ ERROR: null, LEGACY: null }),
+  message: t.string
+});
+
+export type ServicePreferenceError = t.TypeOf<typeof ServicePreferenceError>;
+
+export type ServicePreferenceValueOrError = (params: {
+  readonly serviceId: NonEmptyString;
+  readonly fiscalCode: FiscalCode;
+  readonly userServicePreferencesMode: ServicesPreferencesMode;
+  readonly userServicePreferencesVersion: number;
+}) => TaskEither<
+  ServicePreferenceError,
+  ReadonlyArray<BlockedInboxOrChannelEnum>
+>;
+
+type ServicePreferencesValues = Omit<
+  ServicePreference,
+  "serviceId" | "fiscalCode" | "settingsVersion"
+>;
+
+const channelToBlockedInboxOrChannelEnum: {
+  readonly [key in keyof ServicePreferencesValues]: BlockedInboxOrChannelEnum;
+} = {
+  isEmailEnabled: BlockedInboxOrChannelEnum.EMAIL,
+  isInboxEnabled: BlockedInboxOrChannelEnum.INBOX,
+  isWebhookEnabled: BlockedInboxOrChannelEnum.WEBHOOK
+};
+
+const servicePreferenceToBlockedInboxOrChannels: (
+  servicePreference: ServicePreference
+) => ReadonlyArray<BlockedInboxOrChannelEnum> = servicePreference =>
+  Object.entries(servicePreference)
+    // take only attributes of ServicePreferencesValues
+    .filter(([name, _]) => channelToBlockedInboxOrChannelEnum[name])
+    // take values set to false
+    .filter(([_, isEnabled]) => !isEnabled)
+    // map to BlockedInboxOrChannelEnum
+    .map(([name, _]) => channelToBlockedInboxOrChannelEnum[name]);
+
+const getServicePreferenceValueOrError = (
```

Comment:
If I get it right, the purpose of this subroutine is to reduce the complexity of User's preferences into an array of Services to be blocked. Right?

It would be useful to have a comment which explains that, so that readers do not have to go through inner, technical detail to figure out what the function does.

Reply:
Seems legit.

Reply:
This is my _reposability_, actually :D
and yes, it's exactly what you wrote; rather than writing 3 `if` to check the enabled channels, we decided to do like this. So in case a new channel is added, missing mapping will break the build

Reply:
Rule of thumb: when adding a function, comment **why** you need it